connectors/anaplan/app/clients.py

```
@@ -0,0 +1,98 @@
+
+        # Set default headers for all API calls
+        await self._update_client_headers()
+
+    async def _get_auth_token(self) -> str:
+        """Get authentication token from Anaplan API using basic auth.
+
+        Makes a POST request to Anaplan's authentication endpoint to obtain
+        an authentication token for subsequent API calls.
+
+        Returns:
+            str: The authentication token value.
+
+        Raises:
+            Exception: If authentication fails or token is not found in response.
+
+        Note:
+            Endpoint: POST https://auth.anaplan.com/token/authenticate
+            Expected Response: 201 Created with token information.
+        """
+
+        if self.auth_type == "basic":
+            if not self.username or not self.password:
+                raise Exception(
+                    "Username and password are required for basic authentication"
+                )
+
+            # Pass the auth parameters to BaseClient's execute_http_post_request
+            response = await self.execute_http_post_request(
+                "https://auth.anaplan.com/token/authenticate",
+                auth=(self.username, self.password),
+            )
+
+            if response and response.is_success:  # Success Code : 201
+                token_data = response.json()
+                token_info = token_data.get("tokenInfo", {})
+                self.auth_token = token_info.get("tokenValue")
+                if not self.auth_token:
+                    raise Exception("Token not found in authentication response")
+
+                logger.info("Successfully obtained Anaplan authentication token")
+                return self.auth_token
+            else:
+                error_msg = f"Authentication failed with status {response.status_code if response else 'No response'}: {response.text if response else 'No response'}"
+                logger.error(error_msg)
+                raise Exception(error_msg)
```

I think you should raise a specific exception like `application_sdk.common.error_codes.ClientError` not `Exception`